server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -1311,8 +1324,29 @@ public ClusterState getClusterStateForManifest(
         String localNodeId,
         boolean includeEphemeral
     ) throws IOException {
+
+        ClusterStateTermVersion clusterStateTermVersion = new ClusterStateTermVersion(
+            new ClusterName(clusterName),
+            manifest.getClusterUUID(),
+            manifest.getClusterTerm(),
+            manifest.getStateVersion()
+        );
+        ClusterState lastState = lastDownloadState.get();
+        if (lastState != null) {
+            ClusterStateTermVersion lastStateTermVersion = new ClusterStateTermVersion(
+                new ClusterName(clusterName),
+                lastState.stateUUID(),
+                lastState.term(),
+                lastState.version()
+            );
+            if (clusterStateTermVersion.equals(lastStateTermVersion)) {
+                return lastState;
+            }
+        }
```

Comment:
should we create a RemoteClusterStateCache internal class so that all this code can be moved there and not sprinkled across this class?

Reply:
makes sense. will add a new class

Reply:
Added RemoteClusterStateCache class with get and put